client/app/kubernetes/nodes/kubernetes-nodes.controller.js

```diff
@@ -56,15 +56,16 @@ angular.module('managerApp').controller('KubernetesNodesCtrl', class KubernetesN
   }
 
   getAssociatedFlavor(node) {
-    if (node.instanceId) {
-      return this.Kubernetes.getAssociatedInstance(node.projectId, node.instanceId)
-        .then(instance => _.set(node, 'formattedFlavor', this.Kubernetes.formatFlavor(instance.flavor)))
-        .catch(() => {
-          _.set(node, 'formattedFlavor', this.$translate.instant('kube_nodes_flavor_error'));
-        });
-    }
-
-    return this.$q.when(_.set(node, 'formattedFlavor', node.flavor));
+    return this.Kubernetes.getFlavors(node.projectId)
+      .then((flavors) => {
+        const matchedFlavor = _.find(flavors, flavor => flavor.name === node.flavor);
+        _.set(node, 'formattedFlavor', matchedFlavor
+          ? this.Kubernetes.formatFlavor(matchedFlavor)
+          : this.$translate.instant('kube_nodes_flavor_error'));
```

Comment:
Please use two different messages

Reply:
There would definitely be a matching flavor and hence the error message is removed here.